src/cli/self_update.rs

```diff
@@ -332,7 +332,14 @@ These components can be acquired through a Visual Studio installer.
 
 "#;
 
-static UPDATE_ROOT: &str = "https://static.rust-lang.org/rustup";
+static DEFAULT_UPDATE_ROOT: &str = "https://static.rust-lang.org/rustup";
+
+fn update_root() -> String {
+    process()
+        .var("RUSTUP_UPDATE_ROOT")
+        .inspect(|url| info!("`RUSTUP_UPDATE_ROOT` has been set to `{url}`"))
```

Comment:
Concern: It might not be necessary to print it each time this `update_root` is fetched. Ideally we should print all three altogether at launch time.

Reply:
I think https://github.com/rust-lang/rustup/pull/3722#discussion_r1560041699 has made it less annoying so I'll leave the logic as-is.